app/src/main/java/org/dhis2/data/service/SyncPresenterImpl.kt

```diff
@@ -0,0 +1,368 @@
+package org.dhis2.data.service
+
+import androidx.work.Constraints
+import androidx.work.Data
+import androidx.work.ExistingWorkPolicy
+import androidx.work.ListenableWorker
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequest
+import androidx.work.WorkManager
+import io.reactivex.Completable
+import io.reactivex.Observable
+import java.util.ArrayList
+import java.util.Calendar
+import java.util.concurrent.TimeUnit
+import kotlin.math.ceil
+import org.dhis2.data.prefs.Preference.Companion.DATA
+import org.dhis2.data.prefs.Preference.Companion.EVENT_MAX
+import org.dhis2.data.prefs.Preference.Companion.EVENT_MAX_DEFAULT
+import org.dhis2.data.prefs.Preference.Companion.LIMIT_BY_ORG_UNIT
+import org.dhis2.data.prefs.Preference.Companion.LIMIT_BY_PROGRAM
+import org.dhis2.data.prefs.Preference.Companion.META
+import org.dhis2.data.prefs.Preference.Companion.TEI_MAX
+import org.dhis2.data.prefs.Preference.Companion.TEI_MAX_DEFAULT
+import org.dhis2.data.prefs.Preference.Companion.TIME_DAILY
+import org.dhis2.data.prefs.Preference.Companion.TIME_DATA
+import org.dhis2.data.prefs.Preference.Companion.TIME_META
+import org.dhis2.data.prefs.PreferenceProvider
+import org.dhis2.utils.DateUtils
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.arch.call.D2Progress
+import org.hisp.dhis.android.core.common.State
+import org.hisp.dhis.android.core.imports.TrackerImportConflict
+import org.hisp.dhis.android.core.program.ProgramType
+import timber.log.Timber
+
+class SyncPresenterImpl(
```

Comment:
For next time lets try to have this kind of migration outside testing cycles fixes pls

Reply:
noted 👌